File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/CourseVersionTooNewError.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;

public class CourseVersionTooNewError extends Notification {
  /**
   * Notification to be shown when the plugin is too new to support the current course.
   */
  public CourseVersionTooNewError() {
    super(PluginSettings.A_PLUS, getText("notification.CourseVersionError.title"),
        getText("notification.CourseVersionTooNewError.content"), NotificationType.ERROR);
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/CourseVersionTooNewError.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

/**
 * A notification to be shown when the plugin is too new to support the current course.
 */
class CourseVersionTooNewError : Notification(
    PluginSettings.A_PLUS, MyBundle.message("notification.CourseVersionError.title"),
    MyBundle.message("notification.CourseVersionTooNewError.content"), NotificationType.ERROR
)
